test/example-types/facet.js

```diff
@@ -85,7 +85,7 @@ describe('facet', () => {
       expect(filterAgg).to.deep.equal({
         $match: {
           _id: {
-            $regex: '.*(?=.*dis.*)(?=.*comp.*).*',
+            $regex: '(?=.*dis)(?=.*comp)',
```

Comment:
This doesn't seem right - we need the `.*` at the end of the words or else it would only match suffixes, right? e.g. this wouldn't match `distance` or `computer`

Reply:
@daedalus28 false -- just try it out at regex101.com

Reply:
actually, they're both looking wrong to me now that I try them out myself -- will sort it out

Reply:
<img width="1280" alt="Screenshot 2019-07-03 08 53 10" src="https://user-images.githubusercontent.com/8505584/60606322-fd69a280-9d6f-11e9-97ca-f5bdb62c5a3e.png">
<img width="1280" alt="Screenshot 2019-07-03 08 53 43" src="https://user-images.githubusercontent.com/8505584/60606323-fd69a280-9d6f-11e9-98b2-ea5f1e54d674.png">

Reply:
something screwy is up with the lookaheads, but in general `/abc.*` matches exactly what `/abc/` does

Reply:
@daedalus28 ok, I get it. you're right. The Futil version needs to be replaced with the one we had in bid-search. I'll put in an Futil change

Reply:
<img width="1280" alt="Screenshot 2019-07-03 09 01 31" src="https://user-images.githubusercontent.com/8505584/60606972-2f2f3900-9d71-11e9-9e18-a9661709b610.png">
![Uploading Screenshot 2019-07-03 09.01.37.png…]()

Reply:
ok, so the deal is that they both match the string (which is all we want) but they give different sub-matches. The old mongo version gives better sub-matches than the existing Futil one, so I'll still replace the one in Futil with the old one from here @daedalus28